### DISCUSSION TITLE
Ghost Gremlins?

### DISCUSSION THREAD

**Discussion by team-daniel** [PKU-Alignment/safety-gymnasium#86] [General]
In envs with Gremlins, there are ghost gremlins that move around the environment also. Does anyone know what these are? They also get picked up by the LIDAR that the agent uses which I noticed during training.

I have circled them below in red

![Screenshot from 2023-10-30 09-08-37](https://github.com/PKU-Alignment/safety-gymnasium/assets/55928249/1a17e0d4-ac09-4ea5-9585-27d4f10bb671)

**Comment by muchvo** [comment]
This behavior is due to MuJoCo's Mocap mechanism, which is a feature for creating moving objects. For more details, please refer to [this link](https://mujoco.readthedocs.io/en/latest/modeling.html#mocap-bodies). The ghost-like appearance results from our decision not to set the Mocap object to full transparency. It will not be detected by lidar; only solid objects will. Its primary purpose is to provide an ideal path for the physical object to follow. However, we recognize that this could affect vision-based tasks, potentially simplifying them. We will consider setting it to full transparency to prevent potential issues in the future.
